archive/demultiplexer.go

```diff
@@ -272,7 +272,6 @@ func (receiver *RegularCollectionReceiver) Read(r []byte) (int, error) {
 	// Since we're the "reader" here, not the "writer" we need to start with a read, in case the chan is closed
 	wLen, ok := <-receiver.readLenChan
 	if !ok {
-		close(receiver.readBufChan)
```

Comment:
I'm probably missing something, but why do we no longer close the readBufChan here?

Reply:
This close moves to the `Close` method so that closing is connected to the intent being closing, not triggered from the `!ok` which is the demux side closing.  It's separation of concerns, so that this `Read` only is responsible for reading, if it can.  If the read fails, the channel can close later when the intent closes.

Reply:
I see, thanks